packages/strapi-connector-bookshelf/lib/mount-models.js

```diff
@@ -354,38 +354,34 @@ module.exports = async ({ models, target }, ctx, { selfFinalize = false } = {})
         }
         case 'belongsToMorph':
         case 'belongsToManyMorph': {
-          const association = definition.associations.find(
-            association => association.alias === name
-          );
-
-          const morphValues = association.related.map(id => {
-            let models = Object.values(strapi.models).filter(model => model.globalId === id);
-
-            if (models.length === 0) {
-              models = Object.values(strapi.components).filter(model => model.globalId === id);
-            }
-
-            if (models.length === 0) {
-              models = Object.keys(strapi.plugins).reduce((acc, current) => {
-                const models = Object.values(strapi.plugins[current].models).filter(
-                  model => model.globalId === id
-                );
-
-                if (acc.length === 0 && models.length > 0) {
-                  acc = models;
-                }
-
-                return acc;
-              }, []);
-            }
+          const association = _.find(definition.associations, { alias: name });
+          const morphModelsAndFields = association.related.map(morphModel => {
+            const relatedFields = _.reduce(
+              morphModel.attributes,
+              (fields, attr, attrName) => {
+                const samePlugin =
+                  definition.plugin === attr.plugin ||
+                  (_.isNil(definition.plugin) && _.isNil(attr.plugin));
+                const sameModel = [attr.model, attr.collection].includes(definition.modelName);
+                const isMorph = attr.via === 'related';
```

Comment:
the via attribute might be something else than 'related'. I think you should be using `name` here